src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -638,76 +651,93 @@ event::ptr primitive_inst::realloc_if_needed() {
                    _network.get_engine().is_the_same_buffer(dep_memory(0), output_memory(0))) {
             // Clear out memory if if was previously reused, but now primitive can't be optimized
             _outputs[0] = nullptr;
-            _max_output_layout_count = 0;
+            _max_output_layout_count[0] = 0;
         }
     }
 
     // update layout to ensure that it repsects paddings for correct allocation size
     if (_node_output_layout.data_padding.get_dynamic_pad_dims() != tensor(0)) {
-        size_t rank = updated_layout.get_partial_shape().size();
-        auto current_buf_shape = updated_layout.get_buffer_size().get_partial_shape(rank, std::min(static_cast<size_t>(4), rank));
-        updated_layout = layout(current_buf_shape, updated_layout.data_type, updated_layout.format);
+        size_t rank = updated_layouts[0].get_partial_shape().size();
+        auto current_buf_shape = updated_layouts[0].get_buffer_size().get_partial_shape(rank, std::min(static_cast<size_t>(4), rank));
+        updated_layouts[0] = layout(current_buf_shape, updated_layouts[0].data_type, updated_layouts[0].format);
```

Comment:
@yeonbok @vladimir-paramuzov Would you please explain a bit -
1. why `std::min(static_cast<size_t>(4), rank)` is used? Why 4 here?
2. how does `current_buf_shape` change from `updated_layouts[0].get_partial_shape()`? Do you simply want to add padding to the original shape and assign it to the new shape?